syntax = "proto3";
package alert;





/*
GetAlertCount请求
*/
message GetAlertCountRequest {
    /*
    开始时间
    */
    int32 st = 1 ;
    /*
    结束时间
    */
    int32 et = 2 ;
    /*
    聚合粒度
    */
    string aggregate_interval = 3 ;
    /*
    查询的目标instanceIds(逗号分隔)
    */
    string instanceId__in = 4 ;
    /*
    查询的目标objectIds(逗号分隔)
    */
    string objectId__in = 5 ;
    /*
    查询的目标instanceId
    */
    string instanceId = 6 ;
    /*
    查询的目标objectId
    */
    string objectId = 7 ;
}

/*
GetAlertCount返回
*/
message GetAlertCountResponse {
message Data {
    /*
    聚合周期告警数
    */
    int32 count = 1 ;
    /*
    聚合周期告警ID列表
    */
    repeated string ids = 2 ;
    /*
    聚合周期时间戳
    */
    int32 time = 3 ;
}
        
    
    /*
    返回码
    */
    int32 code = 1 ;
    /*
    返回消息
    */
    string msg = 2 ;
    /*
    统计结果列表
    */
    repeated Data data = 3 ;
}

/*
GetAlertCountApi返回
*/
message GetAlertCountResponseWrapper {
    /*
    返回码
    */
    int32 code = 1 ;
    /*
    返回码解释
    */
    string codeExplain = 2 ;
    /*
    错误详情
    */
    string error = 3 ;
    /*
    返回数据
    */
    alert.GetAlertCountResponse data = 4 ;
}

